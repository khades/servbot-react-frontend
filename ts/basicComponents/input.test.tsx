import * as React from "react";
import { create } from "react-test-renderer";
import Input, { IInputProps } from "./input";

describe("Input", () => {
    it("value change should work", () => {
        let inputValue = "";
        const props: IInputProps = {
            id: "ID",
            getLabel: () => "LABEL",
            getValue: () => inputValue,
            setValue: (value: string) => inputValue = value,
        };

        const input = create(
            <Input {...props} />,
        );

        const inputInstance = input.root.find((el) => el.type === "input");
        inputInstance.props.onChange({ currentTarget: { value: "2" } });
        input.update(
            <Input {...props} />,
        );
        expect(inputValue).toBe("2");
    });
    it("inital value should be properly set", () => {
        const props: IInputProps = {
            id: "ID",
            getLabel: () => "LABEL",
            getValue: () => "testValue",
            setValue: (value: string) => value,
        };

        const input = create(
            <Input {...props} />,
        );

        const inputInstance = input.root.find((el) => el.type === "input");
        expect(inputInstance.props.defaultValue).toBe("testValue");
    });
});
