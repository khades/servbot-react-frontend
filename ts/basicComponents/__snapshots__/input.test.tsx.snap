// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input inital value should be properly set 1`] = `
<div
  className="control-group"
>
  <label
    className="control-group__label"
    htmlFor="ID"
  >
    LABEL
  </label>
  <input
    defaultValue="testValue"
    id="ID"
    onChange={[Function]}
  />
</div>
`;
